### PR TITLE
iso-codes: new package

### PR DESCRIPTION
**Pull request by goxberry**

### CODE REVIEW

File: var/spack/repos/builtin/packages/iso-codes/package.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    url      = "https://deb.debian.org/debian/pool/main/i/iso-codes/iso-codes_4.3.orig.tar.xz"
+
+    version('4.3', sha256='643eb83b2d714e8650ed7112706968d057bf5b101ba71c8ef219e20f1737b141')
+
+    depends_on('m4', type='build')
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool', type='build')
+    depends_on('gettext')
+    depends_on('python@3:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/iso-codes/package.py]: build, link, run?

Comment (goxberry) [reply var/spack/repos/builtin/packages/iso-codes/package.py]: @adamjstewart I'm not sure how to answer that question -- I'd guess `build` only, based on `https://github.com/Homebrew/homebrew-core/blob/master/Formula/iso-codes.rb`, but it probably wouldn't hurt to check the pkgconfig file.

**Comment by goxberry** [comment]
To try to put my flurry of PRs in context: my goal is to package https://github.com/curv3d/curv, which one of my team members needs for CAD/meshing conversions. All of the first-degree dependencies (see https://github.com/curv3d/curv/blob/master/BUILD.md) are in Spack already, except for `gedit`, which has many dependencies of its own.

So far, in a branch, I have commits for the following future PRs:

- `atk` apparently needs `libffi` also, like `pango` does in #12384 -- I think these dependencies are really `type='build'` or `type=('build', 'link')`
- `gobject-introspection` needs a GIR search path by adding `XDG_DATA_DIRS` to a `setup_dependent_environment` method
- `gtk-mac-integration` needs to be added
- `gspell` needs to be added
- `gtksourceview` needs a GIR search path via `XDG_DATA_DIRS` again
- `gsettings-desktop-schema` needs to be added
- `librsvg` needs to be added
- `adwaita-icon-themes` needs to be added
- then `gedit` needs to be added

I can cherry-pick and squash commits as needed to make each of these steps separate PRs. What I could use help with are the following questions:

- Is `setup_dependent_environment` also called when `setup_environment` is called? For example, if I write `my-package` with a `setup_dependent_environment` method that sets environment variables through `run_env`, and then I load `my-package` through, say, Lmod, will `setup_dependent_environment` be called? Or will it only be called if, for example, I load `my-other-package` through Lmod, where `my-other-package` depends on `my-package`?
- Would the spack maintainers be willing to accept a PR that adds a release candidate version of LLVM 9, with the understanding that this version will be replaced with a stable release of LLVM 9 once it exists? I can't actually build `gedit` with `adwaita-icon-themes` because of Python version conflicts in the concretized spec. The `adwaita-icon-themes` package depends on Python 2.x via `librsvg`, `rust`, and `llvm`, but other `gedit` dependencies require Python 3.y.
- Are there any X windows applications in Spack that successfully build from source on macOS? I'm trying to figure out if trying to package `gedit` in Spack for macOS is a fool's errand or if I should expect great difficulties. I can build `gedit` without `adwaita-icon-themes`, but nothing launches, and I don't know if the lack of an X window is due to fundamental issues with linking to X-related libraries on macOS, or some other issue (like the missing icons, or misconfigured X settings on my machine, or mistakes in the `gedit` Spack package).
- Assuming no obvious significant differences between a Homebrew formula and a Spack package (e.g., not a package like `metis`, which is patched in Spack for `parmetis` but not patched for `parmetis` in Homebrew), should I expect to be able to use most installed Homebrew formulas as external packages in Spack? There are some thorny edge cases that seem to cause significant issues -- `intltool` is a good example, because Homebrew will build it with macOS system Perl, which on macOS 10.14.6 is `perl@5.18+cpan`, but installed in a strange way -- the Perl libraries and executable are not installed under a common prefix in the way that Spack or Homebrew would install them. Has anyone had any success with using XQuartz as an external source for various X-related packages in Spack? I have tried this approach without success, and can supply a `packages.yaml` for interested parties in 1-2 hours --  I have a branch in version control with the necessary paths and versions; I just need to filter out all of the extra stuff in my `packages.yaml`.

**Comment by goxberry** [comment]
I'm aiming to cherry-pick out commits and squash them for one or two of these bullets per day, starting tomorrow or Thursday.

**Comment by goxberry** [comment]
One other thing: has the API for adding package-level tests stabilized any more since https://spack.readthedocs.io/en/latest/packaging_guide.html#checking-an-installation was last updated? I'd prefer to add tests as I go to avoid having to debug my submissions by writing more packages that depend on earlier packages I submitted; the lack of tests has led to some embarrassing oversights on my part.

File: var/spack/repos/builtin/packages/iso-codes/package.py (hunk 1 above)
Comment (goxberry) [reply var/spack/repos/builtin/packages/iso-codes/package.py]: @adamjstewart `gettext` and `python@3:` are `'build'` dependencies only. Updated in https://github.com/spack/spack/pull/12389/commits/d8c9fc2036f463d8869fc555b7bddcdb28daadda

**Comment by goxberry** [comment]
Rebased on `develop`.

**Comment by goxberry** [comment]
ping @adamjstewart; this PR should be ready to merge, aside from the failing coverage test

**Comment by adamjstewart** [comment]
Sorry for the delay, feel free to ping me more frequently to remind me to merge!

**Pull request merged**